### PR TITLE
FEATURE: Allow using [quote] bbcode in chat messages

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit enables the "quotes" markdown feature for
chat messages, and also makes sure the quote always has
a link to the quoted post with the force_quote_link
param for PrettyText.

**This core PR must be merged first: https://github.com/discourse/discourse/pull/16034**

### CODE REVIEW

File: app/models/chat_message.rb [hunk 1]
```diff
@@ -121,11 +121,12 @@ def self.uncooked
...
     mentions
     onebox
+    quotes
+    spoiler-alert
+    table
     text-post-process
     upload-protocol
     watched-words
-    table
-    spoiler-alert
```
Comment (nattsw) [inline app/models/chat_message.rb:128]: :D

File: assets/javascripts/discourse/components/chat-live-pane.js [hunk 2]
```diff
@@ -618,6 +618,13 @@ export default Component.extend({
           stagedId: null,
           rateLimited: false,
         });
+
+        // some markdown is cooked differently on the server-side, e.g.
+        // quotes, avatar images etc.
+        if (data.chat_message.cooked !== stagedMessage.cooked) {
+          stagedMessage.set("cooked", data.chat_message.cooked);
+        }
+
```
Comment (nattsw) [inline assets/javascripts/discourse/components/chat-live-pane.js:627]: What would happen if we had just replaced it all the time?

Comment (martin-brennan) [reply assets/javascripts/discourse/components/chat-live-pane.js:627]: I think that would be fine, just didn't want to have to do extra work in Ember land if we didn't have to :)

File: assets/javascripts/discourse/components/chat-live-pane.js [hunk 3]
```diff
@@ -618,6 +618,16 @@ export default Component.extend({
           stagedId: null,
           rateLimited: false,
         });
+
+        // some markdown is cooked differently on the server-side, e.g.
+        // quotes, avatar images etc.
+        if (
+          data.chat_message.cooked &&
+          data.chat_message.cooked !== stagedMessage.cooked
+        ) {
```
Comment (CvX) [inline assets/javascripts/discourse/components/chat-live-pane.js:627]:
```suggestion
        if (data.chat_message.cooked !== stagedMessage.cooked) {
```

**Review by CvX — APPROVED** [review]

File: assets/javascripts/discourse/components/chat-live-pane.js (hunk 3 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/components/chat-live-pane.js:627]: This won't work, sometimes it is undefined or null for whatever reason! :upside_down_face:

Comment (CvX) [reply assets/javascripts/discourse/components/chat-live-pane.js:627]: 👀 I'd like us to know the reason 😅

**Pull request merged**